app/controllers/admin2/listings/order_types_controller.rb

```diff
@@ -0,0 +1,173 @@
+    rescue StandardError => e
+      flash[:error] = e.message
+    ensure
+      redirect_to admin2_listings_order_types_path
+    end
+
+    def destroy
+      shape = ListingShape.find(params[:id])
+      url_name = shape.name
+      raise t('admin2.order_types.errors.cannot_delete_msg', error_msg: shape.delete_shape_msg) unless shape.can_delete_shape?
+
+      @current_community.listings.where(listing_shape_id: shape.id).update_all(open: false, listing_shape_id: nil) # rubocop:disable Rails/SkipsModelValidations
+      deleted_shape = @current_community.shapes.by_name(url_name).first
+      raise t('admin2.order_types.errors.cannot_delete') unless deleted_shape
+
+      deleted_shape.update(deleted: true)
+      flash[:notice] = t('admin2.order_types.successfully_deleted', order_type: t(deleted_shape[:name_tr_key]))
+    rescue StandardError => e
+      flash[:error] = e.message
+    ensure
+      redirect_to admin2_listings_order_types_path
+    end
+
+    def order
+      params[:ids]&.each do |index, object|
+        ListingShape.find(object['id']).update(sort_priority: object['position'])
+      end
+      head :ok
+    end
+
+    def process_summary
+      @process_summary ||= processes.each_with_object({}) { |process, info|
+        info[:preauthorize_available] = true if process.process == :preauthorize
+        info[:request_available] = true if process.author_is_seller == false
+      }
+    end
+
+    def processes
+      @processes ||= TransactionService::API::Api.processes.get(community_id: @current_community.id)[:data]
+    end
+
+    private
+
+    def pick_translation(translations)
+      translations.find { |(locale, _translation)|
+        locale.to_s == I18n.locale.to_s
+      }.second
+    end
+
+    def validate_shape(form)
+      form = Shape.call(form)
+      errors = []
+      if form[:shipping_enabled] && !form[:online_payments]
+        errors << t('admin2.order_types.errors.without_online_payments')
+      end
+      if form[:online_payments] && !form[:price_enabled]
+        errors << t('admin2.order_types.errors.enabled_without_price')
+      end
+      if (form[:units].present? || form[:custom_units].present?) && !form[:price_enabled]
+        errors << t('admin2.order_types.errors.used_without_price')
+      end
+      if errors.empty?
+        Result::Success.new(form)
+      else
+        Result::Error.new(errors.join(", "))
+      end
+    end
+
+    def filter_uneditable_fields(shape, process_summary)
+      uneditable_keys = uneditable_fields(process_summary, shape[:author_is_seller]).select { |_, uneditable| uneditable }.keys
+      shape.except(*uneditable_keys)
+    end
+
+    def common_locals(form:, count:, process_summary:, available_locs:)
+      { uneditable_fields: uneditable_fields(process_summary, form[:author_is_seller]),
+        shape: FormViewLayer.shape_to_locals(form),
+        count: count,
+        harmony_in_use: APP_CONFIG.harmony_api_in_use.to_s == 'true',
```

Comment:
Same as above: I don't think this should be here, as Harmony is no longer used at all and this configuration variable was removed. Is there a reason why it was added here?